app/main/views/g_cloud.py

```diff
@@ -265,3 +276,115 @@ def search_services():
+                                                                  user_email=current_user.email_address,
+                                                                  project_name=request.form.get('project_name',
+                                                                                                'My new project'))
+
+    except HTTPError as e:
+        abort(e.status_code)
+
+    project = api_project['project']
+
+    try:
+        data_api_client.create_direct_award_project_search(user_id=current_user.id,
+                                                           user_email=current_user.email_address,
+                                                           project_id=project['id'],
+                                                           search_url=request.form['search_api_url'])
+    except HTTPError as e:
+        abort(e.status_code)
+
+    flash(PROJECT_CREATED_MESSAGE, 'success')
+
+    return redirect(url_for('.view_project',
+                            project_id=project['id']
+                            ))
+
+
+@direct_award.route('/projects/<int:project_id>', methods=['GET'])
+def view_project(project_id):
+    # Get core data
+    all_frameworks = data_api_client.find_frameworks().get('frameworks')
+    framework = framework_helpers.get_latest_live_framework(all_frameworks, 'g-cloud')
+    content_manifest = content_loader.get_manifest(framework['slug'], 'search_filters')
+    lots_by_slug = framework_helpers.get_lots_by_slug(framework)
+
+    # Get the requested Direct Award Project.
+    project = data_api_client.get_direct_award_project(user_id=current_user.id, project_id=project_id)['project']
+    if not is_direct_award_project_accessible(project, current_user.id):
+        abort(404)
+
+    searches = data_api_client.find_direct_award_project_searches(user_id=current_user.id,
+                                                                  project_id=project['id'])['searches']
+
+    # A Direct Award project has one 'active' search which is what we will display on this overview page.
+    search = list(filter(lambda x: x['active'], searches))[0]
+
+    # We need to get buyer-frontend query params from our saved search API URL.
+    search_query_params = search_api_client.get_frontend_params_from_search_api_url(search['searchUrl'])
+    search_query_params_multidict = MultiDict(search_query_params)
+
+    current_lot_slug = search_query_params_multidict.get('lot', None)
+    filters = filters_for_lot(current_lot_slug, content_manifest, all_lots=framework['lots'])
+    clean_request_query_params = clean_request_args(search_query_params_multidict, filters.values(), lots_by_slug)
+
+    # Now build the buyer-frontend URL representing the saved Search API URL
+    search_page_base_url = url_for('main.search_services')
+    parsed_url = list(urlparse(search_page_base_url))
+    parsed_url[4] = urlencode(search_query_params)
```

Comment:
Could do with a comment here to say what we're expecting to find in [0], [1], [2] and [3]. And might be worth extracting this URL construction into a helper method? It feels a bit dark-artsy and a helper would make this bit directly testable (and more obvious exactly what it's doing).

Reply:
Done.